File: samples/legacy-architecture-samples/simple-stack-example-mvp-view/src/main/java/com/zhuinden/simplestackdemoexamplemvp/util/Preconditions.java
package com.zhuinden.simplestackdemoexamplemvp.util;

/**
 * Created by Owner on 2017. 01. 26..
 */

public class Preconditions {
    public static <T> T checkNotNull(T reference) {
        if(reference == null) {
            throw new NullPointerException();
        } else {
            return reference;
        }
    }

    public static <T> T checkNotNull(T reference, String message) {
        if(reference == null) {
            throw new NullPointerException(message);
        } else {
            return reference;
        }
    }
}

File: simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/util/Preconditions.kt
package com.zhuinden.simplestackdemoexamplefragments.util

/**
 * Created by Zhuinden on 2018. 08. 20.
 */

object Preconditions {
    @JvmStatic
    fun <T> checkNotNull(reference: T?): T = when {
        reference == null -> throw NullPointerException()
        else -> reference
    }

    @JvmStatic
    fun <T> checkNotNull(reference: T?, message: String): T =
        reference ?: throw NullPointerException(message)
}
